src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -177,6 +206,25 @@ async def infer_python_dependencies_via_imports(
         if maybe_disambiguated:
             merged_result.add(maybe_disambiguated)
 
+        if not owners.unambiguous and imp.split(".")[0] not in DEFAULT_UNOWNED_DEPENDENCIES:
+            unowned_imports.add(imp)
+
+    unowned_dependency_behavior = python_infer_subsystem.unowned_dependency_behavior
+    if unowned_imports and unowned_dependency_behavior is not UnownedDependencyUsage.DoNothing:
+        raise_error = unowned_dependency_behavior is UnownedDependencyUsage.RaiseError
+        log = logger.error if raise_error else logger.warning
+        log(
+            f"The following imports in {address} have no owners:\n\n{bullet_list(unowned_imports)}\n\n"
+            "If you are using [python].requirement_constraints, consider adding the relevant package.\n"
+            "Otherwise consider specifying a python_requirement target as a dependency.\n"
+            "See https://www.pantsbuild.org/v2.8/docs/python-third-party-dependencies"
```

Comment:
Can use `doc_url` to construct this based on the current version: see e.g. https://github.com/pantsbuild/pants/blob/c686c9cb1d921241826078d90c24132e7e0bfc55/src/python/pants/backend/docker/target_types.py#L43

Reply:
I see you added the `f` marker. I don't think it'll work until `doc_url` is imported.

FWIW the linked code doesn't use `f` or import `doc_url` so I used it as reference :joy:

Reply:
!!! oy.

Reply:
I'm on mobile, feel free to beat me to the fixes :joy:

Reply:
Apparently I pointed you at the only one in the entire codebase which was broken. Sigh. Thanks!